src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -687,12 +728,32 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
             return allProperties;
         }
 
+        [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
+        private static object? BindParameter(ParameterInfo property, IConfiguration config, BinderOptions options)
+        {
+            string parameterName = property.Name!;
```

Comment:
`ParameterInfo.Name` can be null in [some cases](https://github.com/dotnet/aspnetcore/blob/bc42cade7ce0b1149ab9bdc30c78094b3007fd67/src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs#L1067-L1074). I'm not sure if this can happen for a constructor given manually generated IL. Does anyone know for sure one way or the other?

Reply:
I wouldn't have though so, but I could be wrong. I tried to think of a way how to test this, but couldn't come up with a way that would compile in C#. There might be a cunning way by building expressions dynamically, but I don't know if that'd be a supported mechanism for binding.

Reply:
I checked again in response to another comment in this PR. The docs say that `Name` could be null if the `ParameterInfo` is result of getting a return parameter. But that isn't the case here.

Reply:
The docs don't say that's the only case where `ParaemterInfo` can have a null `Name`. The following will also generate a `Delegate`/`MethodInfo` containing a `ParaemterInfo` which is why I linked to it before. That's why it feels safer to check and throw a better error than a `NullReferenceException` if it happens to be possible to define a constructor with an unnamed parameter.

```C#
var unnamedParameter = Expression.Parameter(typeof(int));
var lambda = Expression.Lambda(Expression.Block(), unnamedParameter);
var ex = Assert.Throws<InvalidOperationException>(() => RequestDelegateFactory.Create(lambda.Compile()));
```